build/Dockerfile

```
@@ -203,14 +204,34 @@ RUN apt-get update \
 	&& apt-get install --no-install-recommends --no-install-suggests -y ca-certificates apt-transport-https wget \
 	&& wget -nv https://github.com/jaegertracing/jaeger-client-cpp/releases/download/${JAEGER_VERSION}/libjaegertracing_plugin.linux_amd64.so -O /usr/local/lib/libjaegertracing_plugin.so
 
+############################################# Build image for Trace downloader #############################################
+FROM opentracing-builder AS tracer-builder
+
+ARG JAEGER_VERSION=0.7.0 
+# Jaeger version matched based on: https://github.com/opentracing-contrib/nginx-opentracing/blob/master/Dockerfile
+
+RUN git clone --branch v${JAEGER_VERSION} https://github.com/jaegertracing/cpp-client.git
+
+WORKDIR /cpp-client/.build
+RUN cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DJAEGERTRACING_WITH_YAML_CPP=ON .. && \
+	make && make install \
+	&& export HUNTER_INSTALL_DIR=$(cat _3rdParty/Hunter/install-root-dir) \
+  	&& cp $HUNTER_INSTALL_DIR/lib/libyaml*so /usr/local/lib/
 
 ############################################# Build image for Opentracing #############################################
 FROM debian as opentracing
 ARG OPENTRACING_VERSION=1.6.0
 
 COPY --from=opentracing-builder /nginx-${NGINX_VERSION}/objs/ngx_http_opentracing_module.so /usr/lib/nginx/modules/ngx_http_opentracing_module.so
 COPY --from=opentracing-builder /usr/local/lib/libopentracing.so.${OPENTRACING_VERSION} /usr/local/lib/libopentracing.so.1
-COPY --from=tracer-downloader /usr/local/lib/libjaegertracing_plugin.so /usr/local/lib/libjaegertracing_plugin.so
+
+# If tracer version is available through tracer downloader
+# COPY --from=tracer-downloader /usr/local/lib/libjaegertracing_plugin.so /usr/local/lib/libjaegertracing_plugin.so
+
+# If tracer version is avilable through tracer builder
+ARG JAEGER_VERSION=0.7.0
+COPY --from=tracer-builder /usr/local/lib/libjaegertracing.so.${JAEGER_VERSION} /usr/local/lib/libjaegertracing_plugin.so
```

```suggestion
COPY --from=tracer-builder /usr/local/lib/libjaegertracing.so.* /usr/local/lib/libjaegertracing_plugin.so
```
Can't we just copy the library with `*` and avoid having JAEGER_VERSION declared in two steps?